airflow-core/src/airflow/traces/tracer.py

```diff
@@ -53,7 +53,7 @@ def add_span(func):
 
     @wraps(func)
     def wrapper(*args, **kwargs):
-        with Trace.start_span(span_name=func_name, component=component):
+        with DebugTrace.start_span(span_name=func_name, component=component):
```

Comment:
Why is `add_span` always setting things as a DebugTrace? If it's only ever used for internal things I think we should rename this to `add_debug_span` to make it clearer

Reply:
Yeah, I don't think that users will ever need to annotate a dag function with the `add_span`. I'll rename it.